hibernate-core/src/main/java/org/hibernate/dialect/pagination/LegacyOracleLimitHandler.java

```diff
@@ -61,6 +61,16 @@ public boolean supportsLimit() {
 		return true;
 	}
 
+	@Override
+	public boolean supportsOffset() {
+		return true;
+	}
+
+	@Override
+	public boolean forceLimitUsage() {
+		return true;
+	}
```

Comment:
In 5.6 we never had an implementation that returned anything other than `false` for `forceLimitUsage`, so can you explain to me why we need this?

Reply:
For Oracle 11, the query ends up being:
`select *
from
    (
        select row_.*,rownum rownum_
        from (
                select id
                from Person
                where name like ?
            ) row_
        where rownum<=?
    )
where rownum_>?`
Hence, a limit parameter is needed, and it is set in `AbstractLimitHandler.bindLimitParameters()`. However for the limit parameter to be set, it needs `hasMaxRows && supportsLimit() || forceLimitUsage();`. Hence, since there is no maxrows indicated, I solved it by overriding forceLimitUsage.
It's either that, or the Oracle 11 dialect should not allow an offset without also indicating a max rows, I think?